### ISSUE TITLE
rpn_loss_box is always equal 0

### ISSUE THREAD

**Issue by cjt222** [eragonruan/text-detection-ctpn#14]
when i train model with MLT data, rpn_loss_box is always equal 0, i want to know if you have change something in lbuild_loss function？ @eragonruan

**Comment by eragonruan** [comment]
@cjt222 yes, I followed the strategy of faster rcnn, you can find it in [proposal_target_layer_tf.py#L72](https://github.com/eragonruan/text-detection-ctpn/blob/1b934c07aa8a3774d38883af83532ec6cf2c7f8c/lib/rpn_msr/proposal_target_layer_tf.py#L72). beacuse the ground truth is also used to calculate the loss, so the phenomenon you observed is normal.

**Comment by cjt222** [comment]
i am sorry to bother you again !!! when you calucate rpn_loss_box like below, why do you multiply 100? it cause rpn_loss_box become large. 
   in fact, i feel total box loss divide number of gt box is right
 rpn_loss_box = 100 * tf.reduce_mean(rpn_loss_box_n) / (tf.reduce_sum(tf.cast(fg_keep, tf.float32)) + 1) @eragonruan

**Comment by eragonruan** [comment]
@cjt222 This is a balance parameter, since the gt is used to calculate the loss, so rpn_loss_box may much smaller than it should be, so I multiply it by a balance parameter. By the way, the total loss function I implemented is not exactly same with the paper. So, If you want, you can rebuild the loss function as mentioned in the origin paper.

**Comment by cjt222** [comment]
what do you mean gt is  is used to calculate the loss? in fact,i remember just like faster rcnn, gt is always used to calculate offset loss between anchor and gt... @eragonruan

**Comment by eragonruan** [comment]
@cjt222 this repo was implemented a few months ago, so I'm not quite sure about this. It seems that there remain bugs about this part, so I'll check it soon.

**Comment by eragonruan** [comment]
@cjt222 I have fixed this. you can checkout the latest codes.
